docs/architecture/adr/ADR043-k8s-resource-access.md

```diff
@@ -0,0 +1,674 @@
+# ADR043: Kubernetes resource access rules
+
+## Status
+
+Pending
+
+## Context
+
+Several different levels of access are required within a kubernetes cluster. The
+GSP uses role based access control so these levels are granted to users and
+groups via roles.
+
+## Decision
+
+We will grant users access to the resources according to the rules given below
+(see Appendix A). Namespaces will be split into "prod" and "non-prod" as the
+access for some resources will be different. Here "prod" and "non-prod" refer to
+classifications of namespace. Examples of "prod" namespaces:
+
+- kube-system
+- istio-system
+- gsp-system
+- verify-doc-checking-build
+- verify-doc-checking-prod
+- verify-proxy-node-build
+- verify-proxy-node-prod
+
+The "non-prod" namespaces are those created for developers to rapidly iterate on
+changes to test they work in a GSP cluster environment and fall completely
+outside the path to release.
+
+Also included in the permissions rules below are those given to the in-cluster
+concourse. These permissions apply to both "prod" and "non-prod" namespaces.
+
+## Consequences
+
+* Site Reliability Engineers (SRE) will be able to effectively operate the GSP
+  while reducing the need to escalate to "admin" level access.
+* service team members (e.g. devs) will be able to iterate changes quickly via
+  branch-based development and/or `kubectl apply` level access to specific
+  namespaces.
+* service team members will not be able to interfere with other devs'
+  namespaces, "system" namespaces or "prod" namespaces.
+* SREs and service team members will not be able to cause service degradations
```

Comment:
aren't we still able to delete pods etc.?

Reply:
Yes, but that won't necessarily cause service degradation (depends on the deployment / replicaset / statefulset configuration and whether there are any pod disruption budgets present). That said, we are allowing deletions of deployments and statefulsets, which we may want to open up for discussion again.

Reply:
Marking resolved as 28 days have passed (even if Christmas was in the middle, that's still a long time)